library/bignum_mod_raw.h

```diff
@@ -170,6 +170,31 @@ void mbedtls_mpi_mod_raw_sub( mbedtls_mpi_uint *X,
                               const mbedtls_mpi_uint *B,
                               const mbedtls_mpi_mod_modulus *N );
 
+/** \brief  Multiply two MPIs, returning the residue modulo the specified
+ *          modulus.
+ *
+ * The size of the operation is determined by \p N. \p A and \p B must have
+ * the same number of limbs as \p N.
+ *
+ * \p X may be aliased to \p A or \p B, or even both, but may not overlap
+ * either otherwise.
+ *
+ * \param[out] X        The address of the result MPI.
+ *                      This must be initialized. Must have enough limbs to
+ *                      store the full value of the result.
+ * \param[in]  A        The address of the first MPI. This must be initialized.
+ * \param[in]  B        The address of the second MPI. This must be initialized.
+ * \param[in]  N        The address of the modulus. Used to perform a modulo
+ *                      operation on the result of the subtraction.
+ * \param[in]  T        The address of an MPI used by the multiplication
+ *                      as a temp variable.
```

Comment:
It isn't clear here how large the temporary storage needs to be. Details could be added, similar to those given for `mbedtls_mpi_core_montmul`.

Reply:
> It isn't clear here how large the temporary storage needs to be. Details could be added, similar to those given for `mbedtls_mpi_core_montmul`.

I believe the equation used in `mbedtls_mpi_core_montmul` is pretty much applicable in more than one cases, including this one. Not directly related to the pr but could we consider it for #6684 ?

Reply:
I am attaching the wording for T:

~~~~~~
 * \param[in,out] T         Temporary storage of size at least 2*AN_limbs+1 limbs.
 *                          Its initial content is unused and
 *                          its final content is indeterminate.
 *                          It must not alias or otherwise overlap any of the
 *                          other parameters.
 ~~~~~~

Reply:
Replying to myself: yes, I could do with this, so doing it in my current PR

Reply:
The information has been added, so resolving this conversation